### PR TITLE
Create and integrate container component for new direct debit form

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you doing in this PR?

This adds a container component to handle state updates to and from the new direct debit form, and updates the form to use Source input components in order to fully integrate the container.

[**Trello Card**](https://trello.com/c/skempebq)

## Why are you doing this?

This brings the direct debit form in line with the presentational and container component pattern we use throughout the new checkout, and also brings the inputs fully into line with the rest of the checkout in terms of their appearance and behaviour.

There will likely need to be some additional work to refine styling and the presentation of error messages, but this sets up all the fundamentals.

## Is this an AB test?
- [ ] Yes
- [x] No

## Accessibility test checklist
 - [x] [Tested with screen reader](https://webaim.org/articles/voiceover/)
 - [x] [Navigable with keyboard](https://www.accessibility-developer-guide.com/knowledge/keyboard-only/browsing-websites/)
 - [x] [Colour contrast passed](https://www.whocanuse.com/)

## Screenshots
Screenshot 2023-09-07 at 11 28 58

### CODE REVIEW

File: support-frontend/assets/components/directDebit/directDebitForm/directDebitForm.tsx [hunk 1]
```diff
@@ -1,258 +1,140 @@
...
+						renderElement={() => props.recaptcha}
+					/>
+				</div>
+			)}
 
-			{formError && (
+			{props.formError && (
 				<div
 					css={css`
 						margin-top: 8px;
```
Comment (rBangay) [inline support-frontend/assets/components/directDebit/directDebitForm/directDebitForm.tsx:115]: could be:

`margin-top: ${space[2]}px;`

File: support-frontend/assets/components/directDebit/directDebitForm/directDebitFormContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	// TODO: This functionality should be moving to a payment button in future
+	void function onSubmit() {
+		void confirmAccountDetails();
+
+		if (recaptchaCompleted) {
+			// void dispatch(payWithDirectDebit(props.onPaymentAuthorisation));
+		} else {
+			dispatch(setFormError(directDebitErrorMessages.notCompletedRecaptcha));
+		}
```
Comment (rBangay) [inline support-frontend/assets/components/directDebit/directDebitForm/directDebitFormContainer.tsx:78]: Do we need the `if else` if we are only using the else?

Comment (i-hardy) [reply support-frontend/assets/components/directDebit/directDebitForm/directDebitFormContainer.tsx:78]: We're not using the entire function; this is left in place just as a rough overview of what needs to happen at the point of form submission.

File: support-frontend/assets/components/directDebit/directDebitForm/directDebitForm.tsx (hunk 1 above)
Comment (i-hardy) [reply support-frontend/assets/components/directDebit/directDebitForm/directDebitForm.tsx:115]: This wasn't added as part of this PR so I think I'll leave this alone for now.

File: support-frontend/assets/components/directDebit/directDebitForm/directDebitForm.tsx [hunk 3]
```diff
@@ -1,258 +1,140 @@
...
-					<ErrorMessage message={formError} svg={<SvgExclamationAlternate />} />
+					<ErrorMessage
+						message={props.formError}
+						svg={<SvgExclamationAlternate />}
+					/>
 				</div>
 			)}
 
-			{/* <PaymentButton onConfirmClick={onSubmit} /> */}
+			{/* <PaymentButton onConfirmClick={props.onSubmit} /> */}
```
[the remark this answers is not in the document]
Comment (i-hardy) [reply support-frontend/assets/components/directDebit/directDebitForm/directDebitForm.tsx:125]: Probably not, but payment/form submission has yet to be implemented for this branch so I'd rather leave this in place for now.

**Review by rBangay — APPROVED** [review]

**Pull request merged**